### PR TITLE
cbuild.sh: add fetch argument

### LINKED ISSUE #138: Add fetch target to cbuild
As proposed in https://github.com/kyx0r/nextvi/issues/115#issuecomment-2466232703. Implementation of a `fetch` "target" for cbuild.sh would make maintenance of a personal fork convenient because the user would be able to pull down changes made from upstream. 

The Makefile patch should also be updated to include this feature. This is a feature I will implement.

### PR DESCRIPTION
**Pull request by unrealapex**
Add fetch argument that merges commits from upstream repository. Things to consider:
- [x] user might have uncommitted changes before trying to merge commit
- [ ] user might want to merge the commits with a different commit name

Closes #138

### CODE REVIEW

**Comment by kyx0r** [comment]
Added what you have so far, also fixed trailing space + tab. If you think the functionality is complete, make sure to close or add more commits going forward

**Comment by unrealapex** [comment]
> make sure to close or add more commits going forward

Would you prefer it if I amended the one feature commit or make separate commits for changes?

**Comment by kyx0r** [comment]
> > make sure to close or add more commits going forward
> 
> Would you prefer it if I amended the one feature commit or make separate commits for changes?

Do any way that is convenient for you, I'll figure it out. Sometimes I need to make some small adjustments to the patch or commit message anyways.

File: cbuild.sh [hunk 1]
```diff
@@ -139,8 +139,21 @@ while [ $# -gt 0 ] || [ "$1" = "" ]; do
...
+        git switch -c upstream-temp
+        git pull https://github.com/kyx0r/nextvi
+        git switch master
+        git rebase --rebase-merges upstream-temp
+        git branch -D upstream-temp
+        log "$G" "Successfully fetched from upstream."
+        ;;
     *)
-        echo "Usage: $0 {install|pgobuild|build|debug|clean}"
+        echo "Usage: $0 {install|pgobuild|build|debug|fetch|clean}"
```
Comment (unrealapex) [inline cbuild.sh:156]: I forgot to update the help message. Fixed in latest commit.

**Pull request closed without merging**